packages/protocol/contracts/L1/hooks/AssignmentHook.sol

```diff
@@ -32,7 +32,7 @@ contract AssignmentHook is EssentialContract, IHook {
         uint64 expiry;
         uint64 maxBlockId;
         uint64 maxProposedIn;
-        bytes32 metaHash;
```

Comment:
emmm since we also allow other implementation of hooks, there is a change others may want to use it?

Reply:
I would opt to keep `metaHash` and put it in the data signed by the prover. Then it's still an option to use it or not, and provers can actually know beforehand what they will have to prove.

Reply:
A combination of `maxBlockId` and `maxProposedIn` could also achieve most of that, but indeed not multiple blocks inside a single data blob.